src/internal/packager/compose.go

```diff
@@ -99,27 +105,36 @@ func getImportedComponent(importComponent types.ZarfComponent) (component types.
 	// Loop over package components looking for a match the componentName
 	for _, componentToCompose := range importedPackage.Components {
 		if componentToCompose.Name == componentName {
-			return *prepComponentToCompose(&componentToCompose, importComponent)
+			everGrowingComposePath = filepath.Join(everGrowingComposePath, importComponent.Import.Path)
+			return *prepComponentToCompose(&componentToCompose, importComponent, everGrowingComposePath)
 		}
 	}
 
 	return component
 }
 
 // Reads the locally imported zarf.yaml
-func getSubPackage(component *types.ZarfComponent) (importedPackage types.ZarfPackage) {
-	path := filepath.Join(component.Import.Path, config.ZarfYAML)
-	utils.ReadYaml(path, &importedPackage)
```

Comment:
@jeff-mccoy This is why it wasn't working. We were still using the relative path from the inner-composed perspective and weren't catching the `err` out of the `ReadYaml` call that was telling us the path was bad 🥲

Reply:
Nice! Quick look on mobile, this looks way cleaner, bless you for using filepath now 😂. I'll dig through it more in a couple hours.  Thanks!!